snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -362,6 +356,34 @@ public FileBasedReceivedSnapshot newReceivedSnapshot(final String snapshotId) {
                         + "'."));
 
     final var directory = buildSnapshotDirectory(parsedSnapshotId);
+    if (directory.toFile().exists()) {
+      actor.run(
+          () -> {
+            try {
+              checkAndCleanupExistingDirectory(snapshotId, parsedSnapshotId, directory);
+              createReceivedSnapshot(parsedSnapshotId, directory, newSnapshotFuture);
+            } catch (final Exception e) {
+              newSnapshotFuture.completeExceptionally(e);
+            }
+          });
+    } else {
+      createReceivedSnapshot(parsedSnapshotId, directory, newSnapshotFuture);
```

Comment:
:question: Isn't there still a race condition if we don't synchronize the complete `newReceivedSnapshot` call? I think it would be safer to just wrap the whole method body in the actor context.

Reply:
Here the assumption was that on raft thread is creating a new received snapshot. So there won't be any race-condition between two creations. But, as you said, I will wrap the whole body in the actor context. The optimization is probably not worth.